pkg/server/server.go

```diff
@@ -203,186 +263,51 @@ func New(ctx context.Context, config *lepconfig.Config, endpoint string, cliUID
-
-		case nvidia_ecc.Name:
-			allComponents = append(allComponents, nvidia_ecc.New(ctx, nvmlInstanceV2))
-
-		case nvidia_memory.Name:
-			allComponents = append(allComponents, nvidia_memory.New(ctx, nvmlInstanceV2))
-
-		case nvidia_gpm.Name:
-			allComponents = append(allComponents, nvidia_gpm.New(ctx, nvmlInstanceV2))
-
-		case nvidia_nvlink.Name:
-			allComponents = append(allComponents, nvidia_nvlink.New(ctx, nvmlInstanceV2))
-
-		case nvidia_power.Name:
-			allComponents = append(allComponents, nvidia_power.New(ctx, nvmlInstanceV2))
-
-		case nvidia_temperature.Name:
-			allComponents = append(allComponents, nvidia_temperature.New(ctx, nvmlInstanceV2))
-
-		case nvidia_utilization.Name:
-			allComponents = append(allComponents, nvidia_utilization.New(ctx, nvmlInstanceV2))
+	gpudInstance := &components.GPUdInstance{
+		RootCtx: ctx,
 
-		case nvidia_processes.Name:
-			allComponents = append(allComponents, nvidia_processes.New(ctx, nvmlInstanceV2))
+		NVMLInstance:         nvmlInstanceV2,
+		NVIDIAToolOverwrites: config.NvidiaToolOverwrites,
 
-		case nvidia_remapped_rows.Name:
-			c, err := nvidia_remapped_rows.New(ctx, nvmlInstanceV2, eventStore)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
-			}
-			allComponents = append(allComponents, c)
-
-		case nvidia_fabric_manager.Name:
-			fabricManagerLogComponent, err := nvidia_fabric_manager.New(ctx, eventStore)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
-			}
-			allComponents = append(allComponents, fabricManagerLogComponent)
-
-		case nvidia_gsp_firmware_mode.Name:
-			allComponents = append(allComponents, nvidia_gsp_firmware_mode.New(ctx, nvmlInstanceV2))
+		Annotations: config.Annotations,
+		DBRO:        dbRO,
 
-		case nvidia_infiniband.Name:
-			c, err := nvidia_infiniband.New(ctx, eventStore, config.NvidiaToolOverwrites)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
-			}
-			allComponents = append(allComponents, c)
+		EventStore:       eventStore,
+		RebootEventStore: rebootEventStore,
 
-		case nvidia_peermem.Name:
-			c, err := nvidia_peermem.New(ctx, eventStore)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
-			}
-			allComponents = append(allComponents, c)
-
-		case nvidia_persistence_mode.Name:
-			allComponents = append(allComponents, nvidia_persistence_mode.New(ctx, nvmlInstanceV2))
-
-		case nvidia_nccl.Name:
-			c, err := nvidia_nccl.New(ctx, eventStore)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create component %s: %w", k, err)
-			}
-			allComponents = append(allComponents, c)
+		MountPoints:  []string{"/"},
+		MountTargets: []string{"/var/lib/kubelet"},
+	}
 
-		case containerd_pod.Name:
-			allComponents = append(allComponents, containerd_pod.New(ctx))
+	allComponents := make([]components.Component, 0)
+	for _, initFunc := range componentInits {
```

Comment:
Shall we merge these three for loop together? L291, L297, L308

Reply:
done